hack/pre-release.sh

```diff
@@ -0,0 +1,193 @@
+#!/bin/bash
+set -euo pipefail
+# this script updates the WMCO and OCP versions in the Makefile, CSV, bundle and Dockerfiles.
+
+# Include the common.sh 
+WMCO_ROOT=$(dirname "${BASH_SOURCE}")/..
+source $WMCO_ROOT/hack/common.sh
+
+CVP=false
+BASEIMAGE=false
+
+# Locate and store the WMCO version from the makefile 
+# finds "WMCO_VERSION ?=" and selects text following it
+WMCO_VERSION=$(sed -n -e 's/^.*WMCO_VERSION ?= //p' Makefile)
+
+usage() {
+  echo "usage:"
+  echo "input the name of the branch you want to increment, and the x y or z value to bump"
+  echo "example: ./pre-release.sh master x"
+  echo "this will bump the x stream of master by 1"
+  echo "optional flags: -c"
+  echo "  bumps the versions held back by CVP"
+  echo "optional flags: -b"
+  echo "  bumps the baseimage versions"
+}
+
+update_WMCO_version() {
+  xyz_stream=$1
+  version=$2 
+  cluster_service_version_yaml=config/manifests/bases/windows-machine-config-operator.clusterserviceversion.yaml
+ 
+  # Setting the stream var to 1 2 or 3 which is used for incrementing the correct stream
+  # this tells awk which column to act on. Column 1, column 2, or column 3.
+  # which correspond to the x, y and z stream respectively. 
+  if [[ $xyz_stream == "x" ]]; then
+    stream_var=1
+  fi
+  if [[ $xyz_stream == "y" ]]; then
+    stream_var=2
+  fi
+  if [[ $xyz_stream == "z" ]]; then
+    stream_var=3
+  fi
+ 
+  # Increment the version
+  updated_version=$(echo $version |
+    awk -F. -v OFS=. -v stream_var="$stream_var" '{$stream_var += 1 ; print}')
+
+  last_version=$(grep "olm.skipRange" $cluster_service_version_yaml |
+    grep -oP ">=\K([0-9]+\.[0-9]+\.[0-9]+)\s*<" | sed "s/\s*<//")
+
+  echo "WMCO version $version detected from the Makefile"
+
+  echo "Updating the Makefile to version $updated_version"
+  # Example: matches *WMCO_VERSION ?= 9.0.0*
+  sed -i "s/WMCO_VERSION ?= $version/WMCO_VERSION ?= $updated_version/g" Makefile
+
+  echo "Updating the skipRange version in $cluster_service_version_yaml to $updated_version"
+  # Example: matches olm.skipRange: '>=8.0.0 *<9.0.0*'
+  sed -i "s/<$version/<$updated_version/g" $cluster_service_version_yaml
+
+  echo "Updating previous skipRange version in $cluster_service_version_yaml to $version"
+  # Example: matches olm.skipRange: '*>=8.0.0* <9.0.0'
+  sed -i "s/>=$last_version/>=$version/g" $cluster_service_version_yaml
+ 
+  echo "Updating bundle/windows-machine-config-operator.package.yaml to $updated_version"
+  # Example: matches   - currentCSV: windows-machine-*config-operator.v9.0.0*
+  sed -i "s/config-operator.v$version/config-operator.v$updated_version/g" bundle/windows-machine-config-operator.package.yaml
+   
+}
+
+update_OCP_version() { 
+  stream=$1
+  WMCO_VERSION=$2 
```

Comment:
Why are you overwriting the global `WMCO_VERSION` variable here? This should just be a local variable.